Sources/GRPC/ConnectionPool/PoolManagerStateMachine+PerPoolState.swift

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import NIO
+
+extension PoolManagerStateMachine.ActiveState {
+  internal struct EventLoopID: Hashable, CustomStringConvertible {
+    private let id: ObjectIdentifier
+
+    internal init(_ eventLoop: EventLoop) {
+      self.id = ObjectIdentifier(eventLoop)
+    }
+
+    internal var description: String {
+      return String(describing: self.id)
+    }
+  }
+
+  internal struct PerPoolState {
+    /// A pool of connections using the same `EventLoop`.
+    internal var pool: ConnectionPool
```

Comment:
Do we need to hold a reference to the actual `ConnectionPool` in the state machine?

Reply:
Could the StateMachine only hold state info, and the `ConnectionPool` itself is hold by the `ConnectionPoolManager`?

Reply:
Maybe we can replace that with the `EventLoopID`.